common/courses/states.ts

```diff
@@ -131,15 +133,19 @@ export async function canCancel(subcourse: Subcourse): Promise<Decision> {
         return { allowed: false, reason: 'already-cancelled' };
     }
 
+    if (isElevated(context)) {
+        return { allowed: true };
+    }
+
     if (await subcourseOver(subcourse)) {
         return { allowed: false, reason: 'subcourse-ended' };
     }
 
     return { allowed: true };
 }
 
-export async function cancelSubcourse(user: User, subcourse: Subcourse) {
-    const can = await canCancel(subcourse);
+export async function cancelSubcourse(user: User, subcourse: Subcourse, context: GraphQLContext) {
```

Comment:
Please only pass an isAdmin boolean to this function, as the functions in common/ are supposed to be as general as possible. The graphql context should be contained in the graphql mutations/resolvers.